lib/BloqadeLattices/test/docstrings.jl

```diff
@@ -0,0 +1,16 @@
+using BloqadeLattices
+using Test
+
+@testset "Docstring Present" begin
+    for symbol in names(BloqadeLattices)
+        obj = Base.eval(@__MODULE__, :(BloqadeLattices.$symbol))
+
+        typeof(obj) <: Union{Module} && continue
+            
+        docs_md = Base.eval(@__MODULE__, :(@doc BloqadeLattices.$symbol))
+        doc_str = docs_md.meta[:results]
+        @testset "$symbol" begin
+            @test length(doc_str) > 0
+        end
+    end
+end
```

Comment:
Let's use the following function instead

```julia
function has_docstring(obj, sig::Type = Union{})
    Docs.getdoc(obj, sig) === nothing || return true
    binding = Docs.Binding(parentmodule(obj), nameof(obj))
    for mod in Docs.modules
        dict = Docs.meta(mod)
        haskey(dict, binding) || continue
        multidoc = dict[binding]
        for msig in multidoc.order
            sig <: msig && return true
        end
    end
    return false
end

has_docstring(sin)
```

Reply:
also maybe just put it in BloqadeCI, and create a CLI entry for it, then we can put it as part of CI. So it doesn't always block the PR.

Reply:
Yeah I think that's more reasonable - we don't have to check this in every PR, sometimes one just want to write a YOLO PR ;-)

Reply:
> ```julia
> function has_docstring(obj, sig::Type = Union{})
>     Docs.getdoc(obj, sig) === nothing || return true
>     binding = Docs.Binding(parentmodule(obj), nameof(obj))
>     for mod in Docs.modules
>         dict = Docs.meta(mod)
>         haskey(dict, binding) || continue
>         multidoc = dict[binding]
>         for msig in multidoc.order
>             sig <: msig && return true
>         end
>     end
>     return false
> end
> ```

This function doesn't seem to be working properly.